core/src/Spirit/Collection.cpp

```diff
@@ -1,9 +1,18 @@
 #include <Spirit/Collection.h>
+#include <utility/Exception.hpp>
 #include <data/State.hpp>
 
 int Collection_Get_NOC(State * state)
 {
-    return state->noc;
+    try
+    {
+        return state->noc;
+    }
```

Comment:
If an invalid `State *` is passed here, `state->noc` may throw. The try/catch should therefore stay.